OpenRA.Game/Server/Server.cs

```diff
@@ -625,11 +678,111 @@ void DispatchOrdersToClient(Connection c, int client, int frame, byte[] data)
 			}
 		}
 
+		// Check if only one side remains
+		bool HasRemainingConflict()
+		{
+			var lastTeam = -1;
+			var remainingPlayers = gameInfo.Players.Where(p => p.Outcome == WinState.Undefined);
+			foreach (var player in remainingPlayers)
+			{
+				if (lastTeam >= 0 && (player.Team != lastTeam || player.Team == 0))
+					return true;
+
+				lastTeam = player.Team;
+			}
+
+			return false;
+		}
+
+		void SetRemainingPlayersAsWinners()
+		{
+			var now = DateTime.UtcNow;
+			var remainingPlayers = gameInfo.Players.Where(p => p.Outcome == WinState.Undefined);
+			foreach (var winner in remainingPlayers)
+			{
+				winner.Outcome = WinState.Won;
+				winner.OutcomeTimestampUtc = now;
+			}
+		}
+
+		void SetPlayerDefeat(int client)
+		{
+			var defeatedPlayer = gameInfo.Players.FirstOrDefault(p => p.ClientIndex == client);
+			if (defeatedPlayer == null || defeatedPlayer.Outcome != WinState.Undefined)
+				return;
+
+			defeatedPlayer.Outcome = WinState.Lost;
+			defeatedPlayer.OutcomeTimestampUtc = DateTime.UtcNow;
+
+			if (!HasRemainingConflict())
+				SetRemainingPlayersAsWinners();
+		}
+
+		bool OutOfSync(int frame)
```

Comment:
OK I'll admit the reason I returned a `bool` was a stylistic concern: to avoid the addition of tons of braces down the road in `HandleSyncOrder()` as you noticed yourself:

```diff
                        {
                                if (packet.Length != existingSync.Length)
-                                       return OutOfSync(frame);
+                               {
+                                       OutOfSync(frame);
+                                       return;
+                               }
                                else
+                               {
                                        for (var i = 0; i < packet.Length; i++)
+                                       {
                                                if (packet[i] != existingSync[i])
-                                                       return OutOfSync(frame);
+                                               {
+                                                       OutOfSync(frame);
+                                                       return;
+                                               }
+                                       }
+                               }
                        }
```

But you also correctly noticed that the else block can be flattened to mitigate the problem, so I just did that.